### ISSUE TITLE
Can't assign custom OnError delegate

### ISSUE THREAD

**Issue by EricBatlle** [ZeShmoutt/Blizzard-API-For-Unity#6]
Hi, 

I'm trying to use the version 1.11 feature: `Added action_onError to all endpoints`

But independently of the delegate passed, it always act as if the `action_OnError` on `APIRequest` is `null`. 
So everytime `DefaultResultAction` is triggered instead of `action_OnError`. 

In this example will popup "Request error : HTTP/1.1 404 Not Found" instead of "Error": 
```
public void Start()
{
    StartCoroutine(BlizzardAPI.WowProfile.GetCharacterProfileSummary(BattleNetRegion.Europe, "zuljin", "arar", (Json_Wow_CharacterProfileSummary) => 
    { 
    	Debug.Log("Ok"); 
    }, null, null, (errorString) => 
    { 
    	Debug.Log("Error"); 
     }));
}
```
I'm doing something wrong?

Also in `DefaultErrorAction` there's an error cause it should be 0 instead of 1 in `Debug.LogErrorFormat("<color=red>Request error :</color> {1}", error);`

**Comment by ZeShmoutt** [comment]
Those are both stupid errors from me, but at least they are easily fixed.

I simply forgot to update the method template to actually include it in the call :
![image](https://user-images.githubusercontent.com/36408045/95778946-69ec8480-0cc9-11eb-9faf-34c7ab94b6b6.png)

I'll work on it when I have some time, work is pretty heavy these days.

-----

On a side note, I heavily suggest that you turn your delegates into separated methods like this :
![image](https://user-images.githubusercontent.com/36408045/95780492-4840cc80-0ccc-11eb-8696-bf8e59c040a4.png)

It's probably not making much of a difference performance-wise, but it's great for readability.

**Comment by EricBatlle** [comment]
> I simply forgot to update the method template to actually include it in the call :
> I'll work on it when I have some time, work is pretty heavy these days.

If you're okey with that I can make a pull request fixing those errors if you're busy these days.

> On a side note, I heavily suggest that you turn your delegates into separated methods like this :
> ![image](https://user-images.githubusercontent.com/36408045/95780492-4840cc80-0ccc-11eb-8696-bf8e59c040a4.png)
> 
> It's probably not making much of a difference performance-wise, but it's great for readability.

Oh, yes, of course, I do It with anonimus functions on the issue just because laziness ^^' but thanks for the tips ZeShmoutt!

**Comment by ZeShmoutt** [comment]
> If you're okey with that I can make a pull request fixing those errors if you're busy these days.

Sure, you could add it on all 200+ endpoints by hand... But I think it would be faster and less painful for you if I can just modify the template and use my custom editor stuff to regenerate everything in a few seconds.

Fixed in version 1.11.1.
